Makefile

```diff
@@ -49,3 +49,42 @@ install-dex: setup-dex-helm-repo ## Install dex on k8s using helm chart
 uninstall-dex: ## Uninstall dex from k8s
 	@helm uninstall -n $(K8S_NS) dex
 	@echo
+
+##@ Certificates:
+
+CA_CERT := certs/ca.crt
+CA_KEY := certs/ca.key
+
+certs:
+	@mkdir -p $@
+
+certs/ca.key: certs
+	@echo Generating $@…
+	@openssl genrsa -out $@ 4096
+
+certs/ca.crt: $(CA_KEY)
+	@echo Generating $@…
+	@openssl req -x509 -new -nodes -days 1825 -sha256 \
+		-key $< -out $@ \
+		-subj "/CN=Dex CA"
+
+certs/%.key: certs
+	@echo Generating $@…
+	@openssl genrsa -out $@ 2048
+
+certs/%.csr: certs/%.key
+	@echo Generating $@…
+	@openssl req -new -key $< -out $@ -nodes -subj "/CN=Dex $*"
+
+certs/%.crt: certs/%.csr $(CA_CERT) $(CA_KEY)
+	@echo Generating $@…
+	@openssl x509 -req -in $< -CA $(CA_CERT) -CAkey $(CA_KEY) -CAcreateserial -days 365 \
+		-out $@
+
```

Comment:
if you dont provide openssl.conf it will try to use default conf it can find in the computer (and fails otherwise). The generated certificate, by default, does not have IP SAN and will fail during handshake if I'm not mistaken.